nixos/modules/services/security/hologram-agent.nix

```diff
@@ -0,0 +1,50 @@
+{pkgs, config, lib, ...}:
+
+with lib;
+
+let
+  cfg = config.services.hologram-agent;
+
+  cfgFile = pkgs.writeText "hologram-agent.json" (builtins.toJSON {
+    host = cfg.dialAddress;
+  });
+in {
+  options = {
+    services.hologram-agent = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the Hologram agent for AWS instance credentials";
+      };
+
+      dialAddress = mkOption {
+        type        = types.str;
+        default     = "localhost:3100";
+        description = "Hologram server and port.";
+      };
+
+      httpPort = mkOption {
+        type        = types.str;
+        default     = "80";
+        description = "Port for metadata service to listen on.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "dummy" ];
+    systemd.services.hologram-agent = {
+      description = "Provide EC2 instance credentials to machines outside of EC2";
+      after       = [ "network.target" ];
+      wantedBy    = [ "multi-user.target" ];
+      preStart = ''
+        /run/current-system/sw/bin/rm -fv /var/run/hologram.sock
+        /run/current-system/sw/bin/ip addr add 169.254.169.254/32 dev dummy0 || true
```

Comment:
I can verify networking.interfaces does not take when declared in config:

>    networking.interfaces.dummy0 = {
>      ipAddress = "169.254.169.254";
>      prefixLength = 32;
>    };

I also tried unused physical device and secondary ip's with ":" syntax.  in all cases, i see the systemd proc exec'd, but [virtual] device is not IP'ed.

> building path(s) ‘/nix/store/ihlgckshi7ig1spdsyccl89b7yiy1an8-unit-network-addresses-enp2s0-1.service’
> building path(s) ‘/nix/store/ivjb7b3brz99aphn90xnkq1ac8rcv80f-unit-network-link-enp2s0-1.service’

so this fails:

`[ERROR  ] 2017-02-02T10:26:39-05:00 (/tmp/nix-build-hologram-20170130-d20d1c3.drv-0/go/src/github.com/AdRoll/hologram/cmd/hologram-agent/main.go:77) Could not start up the metadata interface: listen tcp 169.254.169.254:80: bind: cannot assign requested address`

when IP'ing in `preStart`, it works.